src/serac/physics/utilities/variational_form/array.hpp

```diff
@@ -240,21 +342,45 @@ constexpr auto join(array<T, n1> v1, array<T, n2> v2)
   return joined;
 }
 
+/**
+ * @brief Helper type for iterating over a multidimensional array/tensor
+ * @tparam n The parameter pack of integer indices
+ */
 template <int... n>
 struct IndexSpace {
   static constexpr int size = sizeof...(n);
-  array<char, size>    i;
-  constexpr auto       begin() { return IndexSpace<n...>{}; }
-  constexpr auto       end()
+  /**
+   * @brief The current indices within the array/tensor
+   */
+  array<char, size> i;  // FIXME: Is char too small here?
```

Comment:
IndexSpace is supposed to be for iterating over a small multidimensional array, like for indices in a tensor/array. If those dimensions are getting bigger than 128/256, then tensor and array are likely not the right containers anyway.

Also, `for_constexpr` fills a similar role to IndexSpace, and tends to incur less runtime overhead than this IndexSpace class (but increases compile times, as it forces inlining)

Reply:
I think we should be able to verify this statically (that all the sizes are less than char_max)